core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java

```diff
@@ -85,6 +86,23 @@ public static void register(BuiltinFunctionRepository repository) {
     repository.register(tan());
   }
 
+  /**
+   * Base function for math functions with similar formats that return DOUBLE.
+   *
+   * @param functionName BuiltinFunctionName of math function.
+   * @param formula lambda function of math formula.
+   * @return DefaultFunctionResolver for math functions.
+   */
+  private static DefaultFunctionResolver doubleMathFunctionBase(
```

Comment:
1. You can rename this function now
2. You can use it for `sign`, `sqrt` and  `log(x)` functions too

Reply:
`sign` and `sqrt` now use the base function, `log(x)` format is too different to be included